dependencies.js

```diff
@@ -0,0 +1,364 @@
+"use strict";
+
+/**
+ * @typedef {Object<string, ComponentCategory>} Components
+ * @typedef {{ meta: Object<string, any> } & Object<string, ComponentEntry>} ComponentCategory
+ *
+ * @typedef ComponentEntry
+ * @property {string} [title] The title of the component.
+ * @property {string} [owner] The GitHub user name of the owner.
+ * @property {boolean} [noCSS=false] Whether the component doesn't have style sheets which should also be loaded.
+ * @property {string | string[]} [alias] An optional list of aliases for the id of the component.
+ * @property {Object<string, string>} [aliasTitles] An optional map from an alias to its title.
+ *
+ * Aliases which are not in this map will the get title of the component.
+ * @property {string | string[]} [require]
+ * @property {string | string[]} [modify]
+ * @property {string | string[]} [after]
+ */
+
+var getLoad = (function () {
+
+	/**
+	 * A function which does absolutely nothing.
+	 *
+	 * @type {any}
+	 */
+	var noop = function () { };
+
+	/**
+	 *
+	 * @param {null | undefined | T | T[]} value
+	 * @returns {T[]}
+	 * @template T
+	 */
+	function toArray(value) {
+		if (Array.isArray(value)) {
+			return value;
+		} else {
+			return value == null ? [] : [value];
+		}
+	}
+
+	/**
+	 * Returns a new set for the given string array.
+	 *
+	 * @param {string[]} array
+	 * @returns {StringSet}
+	 * @typedef {Object<string, true>} StringSet
+	 */
+	function toSet(array) {
+		/** @type {StringSet} */
```

Comment:
Should this be a `Hash` vs. a `Set`? Set is a collection of unique values vs a keyed object of booleans.

Reply:
Because we still target ES5 for browsers, I can't use the ES6 Set, so I made a small Set pseudo-type based on JS objects. It's supposed to my substitute for ES6 sets, so I named it that way.

Also, it's not really a _keyed object of booleans_ since there is only one value. The Java HashSet, for example, is implemented as a HashMap where all values are the same object.

Reply:
I'm just talking of the naming of the type here, to be clear, but that makes sense.

Reply:
Ahh, ok. 
I'll respond to the rest of your comments tomorrow. It's almost 2 am where I live.